src/Shapefile.jl

```diff
@@ -324,13 +319,17 @@ function Base.read(io::IO,::Type{Handle})
     mmin = read(io,Float64)
     mmax = read(io,Float64)
     jltype = SHAPETYPE[shapeType]
-    shapes = Vector{jltype}(undef, 0)
+    shapes = Vector{Union{jltype, Missing}}(undef, 0)
```

Comment:
Maybe a matter of taste, but would it be better to start with a Vector{jltype} and re-allocate only when the first NullShape is hit? AFAIK these NullShapes are not that common so for most shape vectors the signatures would remain clean.

Reply:
Hmm yeah, I'm not sure though. Don't have any data on how common NullShapes are, but if we do this, that would also mean that if we want to append new features we cannot append missings, which the spec allows. So I'm afraid this would make it all a bit more complicated, at little gain, since these small unions are faily optimized.

Reply:
Ok, lets stick with this. We shoul;d just be aware that this might be breaking for some existing code so we should make a new minor version when making the next release.

Reply:
Ok, thanks for having a look!